### ISSUE TITLE
Missing mkTemplates folder

### ISSUE THREAD

**Issue by kpedro88** [cms-sw/cmssw#29306]
It looks like a recent change in the build rules ended up dropping the mkTemplates folder used for some utilities:
https://github.com/cms-sw/cmssw-config/blob/8ccbb93b0e24f7f4cd6b147f2b3a2fb739f95fbf/ChangeLog#L11-L12

See https://hypernews.cern.ch/HyperNews/CMS/get/swDevelopment/3554.html for the original report.

**Comment by kpedro88** [comment]
assign core

**Comment by makortel** [comment]
I'm wondering if instead of "fixing" the build rules it would make more sense to move the `mkTemplates` directory e.g. to `FWCore/Skeletons/data`.

**Comment by makortel** [comment]
@smuzaffar What combination of environment variables (or some other mechanism) should I use inside the `mk*` scripts to find a directory `FWCore/Skeletons/mkTemplates` either from the development area, or if not in there from the patch release area, or if not in there from the full release area?

**Comment by smuzaffar** [comment]
@makortel , you can use `$CMSSW_BASE` ( which points to local development area) and if not found then use `$CMSSW_RELEASE_BASE`

**Comment by makortel** [comment]
Do I need to worry about patch releases?

**Comment by smuzaffar** [comment]
No, you don't. Patch releases contians symlinks for missing packages.

**Comment by makortel** [comment]
Excellent, thanks.

Then I'm wondering if we could test these scripts in some way. Now they expect to be run inside a package, e.g. `src/FooPackage`, i.e. a straightforward test would be
```bash
cmsrel $CMSSW_VERSION
cd $CMSSW_VERSION/src
mkdir Test
cd Test
mkedanlzr Analyzer
```
but repeating that within unit tests sounds too much. So maybe the way to go is to add a test flag that allows to run them anywhere.

Including the `clang-tidy` and `clang-format` in the test would be nice as well. What would be an easy way to apply them on a selected subdirectory?

**Comment by makortel** [comment]
Fixed in #29840.
